app/routes/events/components/EventEditor/index.tsx

```diff
@@ -152,8 +153,17 @@ const validate = createValidator({
 });
 
 const EventEditor = () => {
-  const { eventIdOrSlug } = useParams<{ eventIdOrSlug: string }>();
+  let { eventIdOrSlug } = useParams<{
+    eventIdOrSlug: string;
+  }>();
   const isEditPage = eventIdOrSlug !== undefined;
+
+  const { state } = useLocation();
+  // Look for event id in the location state given from the admin "copy event" button
+  if (!isEditPage && state?.id) {
+    eventIdOrSlug = state.id;
+  }
```

Comment:
Is it not possible to use conditional operator when initializing eventIdOrSlug. So moving the if statement predicates to the conditional operator. Conditional operators are a thing of beauty. So in short keeping eventIdOrSlug as a const variable. I am wondering this since you only change it's value once. Shouldn't you use let only if you intend to change the value many times?

Reply:
> Conditional operators are a thing of beauty.

haha yes, at least sometimes.

I totally agree that using `let` is semi bad practice, especially if, as you say, it is not intended to change a lot.

Are you suggesting something like

```ts
  const params = useParams<{
    eventIdOrSlug: string;
  }>();
  const isEditPage = params.eventIdOrSlug !== undefined;
  const { state } = useLocation();
  // Fallback to a potential event id, e.g. given from the admin "copy event" button
  const eventIdOrSlug = params.eventIdOrSlug ?? state?.id;
```

I agree that this is both cleaner and better practice - so unless you were thinking of something else, I'll change it to this. Seems to work completely fine.

Thanks! 😄